ChallengesWithTestsMarkVII/ChallengesSet01.cs

```diff
@@ -8,37 +8,44 @@ public class ChallengesSet01
     {
         public bool AreTwoNumbersTheSame(int num1, int num2)
         {
-            throw new NotImplementedException();
+            return (num1 == num2);          
         }
 
         public double Subtract(double minuend, double subtrahend)
         {
-            throw new NotImplementedException();
+            return (minuend - subtrahend);
         }
 
         public int Add(int number1, int number2)
         {
-            throw new NotImplementedException();
+            
+            return ( number1 + number2);
         }
 
         public int GetSmallestNumber(int number1, int number2)
         {
-            throw new NotImplementedException();
+            return Math.Min(number1, number2); 
         }
 
         public long Multiply(long factor1, long factor2)
         {
-            throw new NotImplementedException();
+            return (factor1 * factor2);
         }
 
         public string GetGreeting(string nameOfPerson)
+        
+
         {
-            throw new NotImplementedException();
+            if (nameOfPerson == "")
+            {
+                return "Hello!";
+            }
+            return $"Hello, {nameOfPerson}!";
         }
-
+   
         public string GetHey()
         {
-            throw new NotImplementedException();
+            return ("HEY!");
```

Comment:
For the others, it's more acceptable to have the unneeded parentheses, but I wouldn't expect to see them around a string.  Consider changing this to just `return "HEY!";`